engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.config.flexible.validators.SettingValueValidator;
+import org.terasology.engine.SimpleUri;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ * @param <T> The type of the value this {@link SettingImpl} contains.
+ */
+public class SettingImpl<T> implements Setting<T> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SettingImpl.class);
+    private final T defaultValue;
+
+    private SimpleUri id;
+
+    private T value;
+
+    private String humanReadableName;
+
+    private String description;
+    private SettingValueValidator<T> validator;
+    private List<PropertyChangeListener> subscribers;
+
+    /**
+     * Creates a new {@link SettingImpl} that does not use a validator with the given id and default value.
```

Comment:
`Creates a new {@link SettingImpl} with the given id and default value but no validator.`